Octokit.Tests.Integration/Helpers/MaintenanceModeContext.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Octokit.Reactive;
+
+namespace Octokit.Tests.Integration.Helpers
+{
+    internal sealed class MaintenanceModeContext : IDisposable
+    {
+        internal MaintenanceModeContext(IConnection connection, bool enabled)
+        {
+            _connection = connection;
+            _initialState = enabled;
+
+            // Ensure maintenance mode is in the desired state
+            EnterpriseHelper.SetMaintenanceMode(_connection, _initialState);
+        }
+
+        private IConnection _connection;
+
+        private bool _initialState;
+        
+        public void Dispose()
+        {
+            // Ensure maintenance mode is restored to desired state
+            EnterpriseHelper.SetMaintenanceMode(_connection, _initialState);
```

Comment:
this is a helper context class to be used via a using statement wrapping a test.  

So a test that sets maintenance ON, needs to ensure that maintenance was OFF to begin with, hence the intial state and setting it in the `ctor`.

But also we dont want to leave GHE in maintenance mode after a test (since that is pretty useless for any subsequent tests) so the destructor should be ensuring maintenance mode is OFF again.  Which make me realise Im currently setting the intial state in destructor but actually should always be setting OFF :grinning: